channel/api/snippets/WorkspaceProvisioning/Program.cs

```diff
@@ -0,0 +1,174 @@
+using Google.Apis.Auth.OAuth2;
+using Google.Api.Gax;
+using Google.Cloud.Channel.V1;
+using Google.Type;
+using System;
+using System.IO;
+using System.Linq;
+
+namespace CodeLab
+{
+    class Program
+    {
+        /***************** REPLACE WITH YOUR OWN VALUES ********************************/
+        private static readonly String jsonPrivateKeyFile = "path/to/json_key_file.json";
+        private static readonly String resellerAdminUser = "admin@yourresellerdomain.com";
+        private static readonly String accountId = "C012345";
+        private static readonly String customerDomain = "example.com";
+        /*******************************************************************************/
+
+        private static readonly String accountParent = "accounts/" + accountId;
+
+        private static CloudChannelServiceClient cloudChannelServiceClient;
+
+        static void Main(string[] args)
+        {
+            GoogleCredential credential;
+
+            using (var stream = new FileStream(jsonPrivateKeyFile, FileMode.Open, FileAccess.Read))
+            {
+                credential =
+                    GoogleCredential.FromStream(stream)
+                        .CreateScoped(new string[] {"https://www.googleapis.com/auth/apps.order"})
+                        .CreateWithUser(resellerAdminUser);
+            }
```

Comment:
Any particular reason you don't use Application Default Credentials specifying that they need to be a  set to a service account credential (for impersonation)?
```suggestion
            GoogleCredential credential = GoogleCredential
                .GetApplicationDefault()
                .CreateScoped(CloudChannelServiceClient.DefaultScopes)
                .CreateWithUser(resellerAdminUser);
```